test/e2e/clusterctl_upgrade_test.go

```diff
@@ -0,0 +1,39 @@
+// +build e2e
+
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo"
+)
+
+var _ = Describe("When testing clusterctl upgrades", func() {
+
+	ClusterctlUpgradeSpec(context.TODO(), func() ClusterctlUpgradeSpecInput {
```

Comment:
the original idea was to setup a context with a timeout, but we never got there because at the end we opted for a set of more granular timeouts.
Might be worth to upgrade across all the tests for better readability

Reply:
Yup, but I assume let's just keep it consistent here and change this via a separate issue+PR

Reply:
Agree this should go in a separated issue/PR